importer/indico_importer/controllers.py

```diff
@@ -36,3 +43,75 @@ def _process(self):
         with plugin.plugin_context():
             data = {'records': importer.import_data(query, size)}
         return jsonify(data)
+
+
+class RHEndTimeBase(RHManageTimetableBase):
+    """Base class for the importer operations"""
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.event_new
+        },
+        'preserved_args': {
+            'importer_name'
+        }
+    }
+
+    @staticmethod
+    def _find_latest_end_dt(entries):
+        latest_dt = None
+        for entry in entries:
+            if latest_dt is None or entry.end_dt > latest_dt:
+                latest_dt = entry.end_dt
+        return latest_dt
+
+    def _format_date(self, date):
+        return date.astimezone(timezone(self.event_new.timezone)).strftime('%H:%M')
+
+
+class RHDayEndTime(RHEndTimeBase):
+    """Get the end_dt of the latest timetable entry or the event start_dt if no entry exist on that date"""
+
+    def _checkParams(self, params):
+        RHEndTimeBase._checkParams(self, params)
+        self.date = timezone(self.event_new.timezone).localize(datetime.strptime(request.args['selectedDay'],
+                                                                                 '%Y/%m/%d'))
+
+    def _process(self):
+        end_date = self.date + relativedelta(days=1)
+        entries = self.event_new.timetable_entries.filter(TimetableEntry.start_dt >= self.date,
```

Comment:
`self.date` is in the event timezone but the columns are in utc. You probably need `TimetableEntry.start_dt.astimezone(self.event_new.tzinfo)`